gkr_iop/src/gkr.rs

```diff
@@ -131,21 +132,18 @@ impl<E: ExtensionField> GKRCircuit<E> {
             )?;
         }
 
-        Ok(GKRClaims(
-            self.opening_evaluations(&evaluations, &challenges),
-        ))
+        Ok(GKRClaims(self.opening_evaluations(&evaluations)))
     }
 
-    fn opening_evaluations(
-        &self,
-        evaluations: &[PointAndEval<E>],
-        challenges: &[E],
-    ) -> Vec<Evaluation<E>> {
-        self.openings
+    /// Output opening evaluations. First witin and then fixed.
+    fn opening_evaluations(&self, evaluations: &[PointAndEval<E>]) -> Vec<Evaluation<E>> {
+        let input_layer = self.layers.last().unwrap();
+        input_layer
+            .in_eval_expr
             .iter()
+            .enumerate()
             .map(|(poly, eval)| {
-                let poly = *poly;
-                let PointAndEval { point, eval: value } = eval.evaluate(evaluations, challenges);
+                let PointAndEval { point, eval: value } = evaluations[*eval].clone();
                 Evaluation { value, point, poly }
```

Comment:
Does 3rd argument: `poly` necessary? asking this because in `Vec<Evaluation>`, the poly index are well order: 0, 1, 2, 3, .... So it seems we can retrieve the information from its index in evaluation vector